### PR TITLE
Fixes #258 : Server connection Error shows custom messages

### LINKED ISSUE #258: Server connection Error message should provide more context
On trying to connect to the server through the app, error message should provide more context instead of just saying "Failed to fetch backends"

### PR DESCRIPTION
**Pull request by saketkumar95**
This PR addresses [#258](https://github.com/zulip/zulip-android/issues/258)

Now custom messages shows on server errors like:

- Not connected to internet
- Server doesn't exists on that URL
- Text written there is not a valid URL

@timabbott @kunall17 Review and let me know if there any changes required. :)

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@saketkumar95, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by saketkumar95** [comment]
@kunall17 Review now.

File: app/src/main/java/com/zulip/android/activities/LoginActivity.java [hunk 1]
```diff
@@ -193,60 +198,83 @@ private void showBackends(String httpScheme, String serverURL) {
             serverURL = serverURL + "/";
         }
 
-        Uri serverUri = Uri.parse(serverURL);
+        isValid = URLUtil.isValidUrl(String.valueOf(serverURL));
 
-        serverUri = serverUri.buildUpon().scheme(httpScheme).build();
+        if (!isValid){
+            Toast.makeText(LoginActivity.this, R.string.invalid_url , Toast.LENGTH_SHORT).show();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/LoginActivity.java]: No need to create the else condition, just put the return;

File: app/src/main/java/com/zulip/android/activities/LoginActivity.java [hunk 2]
```diff
@@ -56,6 +60,7 @@
     private EditText mUserName;
     private EditText mPassword;
     private EditText serverIn;
+    private boolean isValid;
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/LoginActivity.java]: Can be removed from a global variable

**Comment by saketkumar95** [comment]
@kunall17 Done! :)

**Comment by kunall17** [comment]
@timabbott LGTM!

**Comment by timabbott** [comment]
@saketkumar95 can you fix the merge conflict?

**Comment by saketkumar95** [comment]
@timabbott Done! :)

**Comment by timabbott** [comment]
I just cleaned this up and merged, but in the future @saketkumar95 please use `git pull --rebase` to resolve merge conflicts, not `git pull` -- Zulip as a project does not use merge commits.

**Comment by saketkumar95** [comment]
@timabbott Yeah sure, Will keep check of it. Thanks :)

**Comment by timabbott** [comment]
I guess while we're talking about this stuff, you may want to check out the Zulip commit message guidelines for details on how we like commit messages written: https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages.  Usually I make everyone fix commit messages in the main project, though we've been a bit more lose about it in the Android app; but we should at least have regular contributors writing them using the standard style (our style is the same as for other major projects using Git like the Linux kernel).

**Pull request closed without merging**